src/cli/main.c

```diff
@@ -33,19 +33,99 @@
 #include "common/image.h"
 #include "common/image_cache.h"
 #include "common/imageio.h"
+#include "common/imageio_jpeg.h"
 #include "common/imageio_module.h"
 #include "common/exif.h"
 #include "common/history.h"
+#include "control/conf.h"
+#include "develop/imageop.h"
 
 #include <sys/time.h>
 #include <unistd.h>
 #include <inttypes.h>
 #include <libintl.h>
 
+static void generate_thumbnail_cache()
+{
+  const int max_mip = DT_MIPMAP_2;
+  fprintf(stderr, _("creating cache directories\n"));
+  char filename[PATH_MAX] = {0};
+  for(int k=DT_MIPMAP_0;k<=max_mip;k++)
+  {
+    snprintf(filename, PATH_MAX, "%s.d/%d", darktable.mipmap_cache->cachedir, k);
+    fprintf(stderr, _("creating cache directory '%s'\n"), filename);
+    int mkd = g_mkdir_with_parents(filename, 0750);
+    if(mkd)
```

Comment:
yes. so the code is correct, right? and if(mkd) is shorter than if(mkd == -1) .. also less error prone in case the api is extended by another error code.